servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/DefaultClientSslConfigBuilder.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.TrustManagerFactory;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default builder for {@link ClientSslConfig} objects.
+ */
+public final class DefaultClientSslConfigBuilder extends AbstractSslConfigBuilder<DefaultClientSslConfigBuilder> {
+    /**
+     * See <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#jssenames">
+     * Endpoint Identification Algorithm Name</a>.
+     */
+    @Nullable
+    private String hostnameVerificationAlgorithm = "HTTPS";
+    private String peerHost = "";
+    private int peerPort = -1;
+    @Nullable
+    private String sniHostname;
+
+    /**
+     * Create a new instance using this JVM's {@link TrustManagerFactory#getDefaultAlgorithm()} and default
+     * {@link TrustManagerFactory}.
+     */
+    public DefaultClientSslConfigBuilder() {
+    }
+
+    /**
+     * Create a new instance using {@code tmf} to verify trusted servers.
+     * @param tmf The {@link TrustManagerFactory} used to verify trusted servers.
+     */
+    public DefaultClientSslConfigBuilder(TrustManagerFactory tmf) {
```

Comment:
some form of trust manager is "required" it just so happens the default constructor falls back to the JVM's default trust manager, which may not be appropriate for all use cases (e.g. default trust manager emulates web browser behavior if not overridden). Due to its relative priority I think it is worth calling it out explicitly rather than having it blend in with the other setters (especially when contrasted against the server side that requires the key manager in the constructor).